src/Sylius/Bundle/ApiBundle/Serializer/Normalizer/ShippingMethodNormalizer.php

```diff
@@ -23,67 +25,56 @@
 use Sylius\Component\Core\Repository\ShipmentRepositoryInterface;
 use Sylius\Component\Registry\ServiceRegistryInterface;
 use Symfony\Component\HttpFoundation\RequestStack;
-use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
 use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
 use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 use Webmozart\Assert\Assert;
 
-final class ShippingMethodNormalizer implements ContextAwareNormalizerInterface, NormalizerAwareInterface
+final class ShippingMethodNormalizer implements NormalizerInterface, NormalizerAwareInterface
 {
     use NormalizerAwareTrait;
 
     private const ALREADY_CALLED = 'sylius_shipping_method_normalizer_already_called';
 
     public function __construct(
+        private SectionProviderInterface $sectionProvider,
         private OrderRepositoryInterface $orderRepository,
         private ShipmentRepositoryInterface $shipmentRepository,
         private ServiceRegistryInterface $shippingCalculators,
         private RequestStack $requestStack,
-        private ChannelContextInterface $channelContext,
     ) {
     }
 
     public function normalize($object, $format = null, array $context = [])
     {
         Assert::isInstanceOf($object, ShippingMethodInterface::class);
         Assert::keyNotExists($context, self::ALREADY_CALLED);
+        Assert::isInstanceOf($this->sectionProvider->getSection(), ShopApiSection::class);
```

Comment:
Do we need this check here if we check the same in `supportsNormalization()`?

Reply:
That's why it's an assertion and not just a check, to be sure I prefer to verify it